### PR TITLE
Feat/459655  validate jwt token

### PR DESCRIPTION
**Pull request by whitewaterdesign**

### CODE REVIEW

File: app/config/auth.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  }
+})
+
+// Define config schema
+const schema = Joi.object({
+  authTokens: Joi.object({
+    portalKey: Joi.string().required(),
+    enforcementKey: Joi.string().required()
+  }),
+  permittedDomains: jsonArray.array().items(Joi.string().pattern(/^[a-z0-9.@]+([-.][a-z0-9]+)*\.[a-z]{2,6}$/)).default([])
```
Comment (jbarnsley10) [inline app/config/auth.js:30]: Nice - I see you have a test for this regex?

File: app/routes/external-events.js [hunk 2]
```diff
@@ -1,10 +1,12 @@
 const { getExternalEvents } = require('../repos/external-events')
 const { externalEventsQuerySchema, externalEventsResponseSchema } = require('../schemas/external-events')
+const { scopes } = require('../constants/auth')
 
 module.exports = {
   method: 'GET',
   path: '/external-events',
   options: {
+    auth: { scope: [scopes.admin] },
```
Comment (jbarnsley10) [inline app/routes/external-events.js:9]: Did you miss the auth scope for /events, or did it not need anything?

Comment (whitewaterdesign) [reply app/routes/external-events.js:9]: It doesn't need auth scope as will allow all valid scopes

File: helm/aphw-ddi-events/values.yaml [hunk 3]
```diff
@@ -37,6 +37,9 @@ container:
   azureStorageConnectionString: dummy
   azureStorageUseConnectionString: false
   azureStorageAccountName: dangerousdogindex
+  publicKeyPortal: dummy
+  publicKeyEnforcement: dummy
+  permittedDomains: "[]"
```
Comment (jbarnsley10) [inline helm/aphw-ddi-events/values.yaml:42]: Not sure if the pipeline likes double-quotes in the yaml. Don;t think I've used them before so might just need to see if ti gets through

**Review by jbarnsley10 — APPROVED** [review]
Looks good. A couple of comments for clarification

File: app/config/auth.js (hunk 1 above)
Comment (whitewaterdesign) [reply app/config/auth.js:30]: am testing it in test/unit/config/auth.test.js

Comment (whitewaterdesign) [reply app/config/auth.js:30]: should fail if PERMITTED_DOMAINS has a value which is not a domain

Comment (jbarnsley10) [reply app/config/auth.js:30]: Haha - sorry - didn't mean to add a question mark. I wasn't asking, I was stating :)

**Pull request merged**